subprojects/docs/src/docs/userguide/working_with_files.adoc

```diff
@@ -334,14 +334,17 @@ See how we're using the `compileJava` task as the source of the files to package
 Using a task directly as an argument like this relies on it having <<more_about_tasks.adoc#sec:task_inputs_outputs,defined outputs>>, so it won't always be possible. In addition, this example could be improved further by relying on the Java plugin's convention for `destinationDir` rather than overriding it, but it does demonstrate the use of project properties.
 
 [[sec:single_file_paths]]
-=== Single file paths
+=== Single file
 
-One of the great quandaries when developing a build is how to specify file locations when the build may be executed from an arbitrary directory — not necessarily in the project — and may be run on any number of different systems with incompatible directory layouts. The standard Java mechanism for specifying a file path runs into trouble in these situations:
+Gradle provides the link:{groovyDslPath}/org.gradle.api.Project.html#org.gradle.api.Project:file(java.lang.Object)[Project.file(java.lang.Object)] method for specifying the location of a single file or directory.
+Relative paths are resolved relative to the project directory, absolute paths remain unchanged.
 
- * `new File(relative path)` generates a path relative to the current working directory, which could be anywhere
- * `new File(absolute path)` will fail if the file system doesn't have the requisite path.
+[CAUTION]
+====
+Never use `new File(relative path)`, as this creates a path relative to the current working directory, which could be anywhere.
```

Comment:
Is "never" too strong here? Is it worth qualifying it with "unless you explicitly do want this behaviour" or ", which could be anywhere and is rarely what you want."?

Reply:
It's literally never correct. The working directory is set in stone when the daemon is started, it can't be changed later. It may in fact be a good idea to set the daemon's working directory to a random temp dir just to make this very clear. Maybe something to do for 5.0

Reply:
So that effectively means that even thinking in terms of the current working directory is bad/dangerous/wrong? Because there's no guarantee that it's even where the user is executing the build from? I think that should be called out explicitly.

If all that is the case, I'd argue that the current wording of the caution is confusing. It says that `new File(path)` creates a path relative to the current working directory, but it's not clear to the reader that "current working directory" probably isn't what they think it is.

Reply:
There was a guarantee until Java 11, but there no longer is. But even if that still worked, it just doesn't make sense. Why would a build ever generate something into the current working dir, which has nothing to do with that build's structure at all?